### PR TITLE
[BCE-23530] added mechanism for fetching description for sidebar

### PR DESCRIPTION
**Pull request by PavloYakovenko**

### CODE REVIEW

File: src/views/interface/checkovResult/webviewPanel.ts [hunk 1]
```diff
@@ -147,6 +150,33 @@ export class CheckovResultWebviewPanel {
...
+                }
+        } catch (e: any) {
+            if (retryCount === 3) {
+                throw new Error('Unable to fetch description: ' + e.message);
+            }
+
+            if (e.response.status === 403) {
+                retryCount++;
+                await AnalyticsService.setAnalyticsJwtToken();
+                return await CheckovResultWebviewPanel.fetchDescription(checkId);
```
Comment (bc-vitrof) [inline src/views/interface/checkovResult/webviewPanel.ts:172]: Isn't it infinite recursion, shouldn't we pass the `retryCount` as a parameter? I think the `retryCount` will be always set to 0 on call.

Comment (PavloYakovenko) [reply src/views/interface/checkovResult/webviewPanel.ts:172]: agree

Comment (PavloYakovenko) [reply src/views/interface/checkovResult/webviewPanel.ts]: fixed

File: src/views/interface/checkovResult/webviewPanel.ts [hunk 2]
```diff
@@ -147,6 +151,33 @@ export class CheckovResultWebviewPanel {
...
+            if (e.response.status === 403) {
+                CheckovResultWebviewPanel.retryCount++;
+                await AnalyticsService.setAnalyticsJwtToken();
+                return await CheckovResultWebviewPanel.fetchDescription(checkId);
+            }
+
+            console.log('Error: ' + e.message, e);
+            return;
+        }
+    }
```
Comment (bc-vitrof) [inline src/views/interface/checkovResult/webviewPanel.ts:179]:
```suggestion
    public static async fetchDescription(checkId: string, retryCount = 0): Promise<string | undefined> {
        const jwtToken = AnalyticsService.applicationContext.globalState.get(GLOBAL_CONTEXT.JWT_TOKEN) as string;

        try {
            const response = await axios.get(CheckovResultWebviewPanel.getDescriptionsEndpoint(checkId), { headers: {
                'Authorization': jwtToken } });
            
                if (response.status === 200) {
                    return response.data.description;
                }
        } catch (e: any) {
            if (retryCount === 3) {
                throw new Error('Unable to fetch description: ' + e.message);
            }

            if (e.response.status === 403) {
                await AnalyticsService.setAnalyticsJwtToken();
                return await CheckovResultWebviewPanel.fetchDescription(checkId, retryCount + 1);
            }

            console.log('Error: ' + e.message, e);
            return;
        }
    }
```

**Review by bc-vitrof — APPROVED** [review]

**Pull request merged**